src/cli/self_update.rs

```diff
@@ -853,6 +853,30 @@ fn install_proxies_with_opts(process: &Process, force_hard_links: bool) -> Resul
     Ok(())
 }
 
+fn check_proxy_sanity(process: &Process, components: &[&str]) -> Result<()> {
+    let bin_path = process.cargo_home()?.join("bin");
+
+    // Sometimes linking a proxy produces an unpredictable result, where the proxy
+    // is in place, but manages to not call rustup correctly. One way to make sure we
+    // don't run headfirst into the wall is to at least try and run our freshly
+    // installed proxies, to see if they return some manner of reasonable output.
+    for component in components {
```

Comment:
I believe it should be expressed through this code that component names and binary names are not necessarily identical, but if you limit it to a representative portion like `rustc` and `cargo` then they happen to collide, but that also has to go into the comments. BTW I don't particularly like the idea of checking all components, maybe just `rustc` and `cargo` would be enough?

Reply:
Sure, I can limit it to those two, provided they've actually been installed.